test/CMakeLists.txt

```diff
@@ -156,7 +156,7 @@ target_link_libraries(LimitedAllocatorTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
 
 add_executable(MinusTest MinusTest.cpp)
 add_test(MinusTest MinusTest)
-target_link_libraries(MinusTest gtest_main engine ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(MinusTest gtest_main engine ${CMAKE_THREAD_LIBS_INIT} boost_iostreams)
```

Comment:
Can you link `engine` directly against `boost_iostreams`, getting rid of most of the explicit linkings here?

Also, (this is my mistake) in the main `CMakeLists.txt` don't link the `ServerMain` against all boost libs, but only against `program_options` (and `iostreams`, if my suggestion from above does not work)